lib/internal/quic/core.js

```diff
@@ -1858,7 +1858,6 @@ class QuicSession extends EventEmitter {
 
     if (!this[kHandshakePost]()) {
       if (typeof state.handshakeCompletePromiseReject === 'function') {
-        // TODO(@jasnell): Proper error
         state.handshakeCompletePromiseReject(
           new ERR_OPERATION_FAILED('Handshake failed'));
       }
```

Comment:
Can I disagree about these TODOs being irrelevant now? `ERR_OPERATION_FAILED` is too generic imo, because an operation failing is just about the definition of an error :)

Reply:
I know we've had a conversation about this before when you mentioned that our errors are too specific in cases. Struggling to find the right balance. Do you have a more concrete suggestion?